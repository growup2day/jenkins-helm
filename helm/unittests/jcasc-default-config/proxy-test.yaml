templates:
  - config-jcasc.yaml
tests:
  - it: should include proxy config in default config scripts when configured
    set:
      JCasC:
        defaultConfig: true
        master:
          proxy:
            name: "10.1.1.1"
            port: 80
            noProxyHost: |-
              *.service.dev
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-proxy.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-proxy.yaml]
  - it: should include blank proxy config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-proxy.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-proxy.yaml]
  - it: should not include proxy config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            jenkins:
              proxy:
                name: 10.1.1.10
    asserts:
      - isNull:
          path: data.[jcasc-default-config-proxy.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
