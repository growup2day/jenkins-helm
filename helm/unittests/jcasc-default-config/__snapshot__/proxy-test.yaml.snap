should include blank proxy config in default config scripts when not configured:
  1: |
    |-
      jenkins:
        # Manage Plugins > Advanced
        proxy:
should include proxy config in default config scripts when configured:
  1: |
    |-
      jenkins:
        # Manage Plugins > Advanced
        proxy:
          name: 10.1.1.1
          noProxyHost: '*.service.dev'
          port: 80
